cluster-autoscaler/cloudprovider/oci/nodepools/oci_manager.go

```diff
@@ -249,11 +250,15 @@ func (m *ociManagerImpl) TaintToPreventFurtherSchedulingOnRestart(nodes []*apiv1
 }
 
 func (m *ociManagerImpl) forceRefresh() error {
-	err := m.nodePoolCache.rebuild(m.staticNodePools)
+	httpStatusCode, err := m.nodePoolCache.rebuild(m.staticNodePools, maxGetNodepoolRetries)
 	if err != nil {
+		if httpStatusCode == 404 {
```

Comment:
If the OCI SDK returns a 404, is that considered an error?

Reply:
Yes, in case of 404, err is not nil.